### PR TITLE
Move trigger handling from PLPGSQL to C

### PR DESCRIPTION
**Pull request by cevian**
Applying triggers to chunks requires taking the definition
of a trigger on a hypertable and executing it on a chunk. Previously
this was done with string replacement in the trigger definition.
This was not especially safe, and thus we moved the logic to C
where we can do proper parsing/deparsing and replacement of the table
name. Another positive aspect is that we got rid of some DDL triggers.

### CODE REVIEW

File: src/process_utility.c [hunk 1]
```diff
@@ -314,6 +320,94 @@ process_drop(Node *parsetree)
 	return handled;
 }
 
+typedef struct DropTriggerCtx
```
Comment (erimatnor) [inline src/process_utility.c]: You don't need this if only passing a single value/pointer. Just pass the `trigger_name` directly instead.

File: src/process_utility.c [hunk 2]
```diff
@@ -536,6 +630,58 @@ process_altertable(Node *parsetree)
...
+		.queryString = query_string,
+		.hypertable_relid = relid
+	};
+
+	if (!stmt->row)
+		return;
+
+	if (!OidIsValid(relid))
+		return;
+
```
Comment (erimatnor) [inline src/process_utility.c]: Unnecessary whitespace here.

File: src/process_utility.c [hunk 3]
```diff
@@ -536,6 +630,58 @@ process_altertable(Node *parsetree)
...
+	CreateIndexCtx ctx = {
+		.stmt = stmt,
+		.queryString = query_string,
+		.hypertable_relid = relid
+	};
+
+	if (!stmt->row)
+		return;
+
+	if (!OidIsValid(relid))
```
Comment (erimatnor) [inline src/process_utility.c]: Merge the two if clauses here?

File: src/process_utility.c [hunk 4]
```diff
@@ -536,6 +630,58 @@ process_altertable(Node *parsetree)
...
+		return;
+
+
+	hcache = hypertable_cache_pin();
+
+	ht = hypertable_cache_get_entry(hcache, relid);
```
Comment (erimatnor) [inline src/process_utility.c]: There is a convenience function `is_hypertable()` that you can use if you only need to know that the relation is a hypertable and you don't need the hypertable info. That allows you to get rid of the cache pinning as well.

File: src/process_utility.c [hunk 5]
```diff
@@ -314,6 +320,94 @@ process_drop(Node *parsetree)
...
+		trigger = trigger_by_oid(address.objectId, stmt->missing_ok);
+
+		if (trigger_is_chunk_trigger(trigger))
+		{
+			Hypertable *ht;
+			DropTriggerCtx ctx = {
+				.trigger_name = trigger_name(trigger)
+			};
+
+			ht = hypertable_cache_get_entry(hcache, relation->rd_id);
```
Comment (olofr) [inline src/process_utility.c]: Maybe use `is_hypertable` here as well as @erimatnor mention below.
